llvm/lib/Target/X86/X86InstrInfo.cpp

```diff
@@ -3147,8 +3148,9 @@ X86::CondCode X86::getCondFromSETCC(const MachineInstr &MI) {
 }
 
 X86::CondCode X86::getCondFromCMov(const MachineInstr &MI) {
-  return X86::isCMOVCC(MI.getOpcode()) ? X86::getCondFromMI(MI)
-                                       : X86::COND_INVALID;
+  return X86::isCMOVCC(MI.getOpcode()) || X86::isCFCMOVCC(MI.getOpcode())
```

Comment:
CFCMOV should not be added here b/c CMOV optimization is not always applicable to CFCMOV.